### ISSUE TITLE
Add asyncio.shield around AsyncSession.close

### ISSUE THREAD

**Issue by dolamroth** [DmitryBurnaev/starlette-web#59]
```
The garbage collector is trying to clean up connection <AdaptedConnection <asyncpg.connection.Connection object at ...>>. This feature is unsupported on async dbapi, since no IO can be performed at this stage to reset the connection. Please close out all connections when they are no longer used, calling ``close()`` or using a context manager to manage their lifetime.
```

The recommendation for handling this issue is https://stackoverflow.com/a/74221652

This will also allow to vary a class AsyncSession via settings. (i.e. to allow to add auto commit/rollback at end, if user wants so)

**Comment by dolamroth** [comment]
Also add
poolclass=AsyncAdaptedQueuePool

Limit constance database backend to pool_size=1, max_overflow=0

**Comment by dolamroth** [comment]
Fix is already introduces in SQLAlchemy 1.4.42, updated minimum version in setup.cfg
